slamd/discovery/processing/discovery_service.py

```diff
@@ -123,3 +127,23 @@ def _initialize_experiment(cls, dataframe, request_body):
             apriori_thresholds=apriori_thresholds,
             apriori_max_or_min=apriori_max_or_min,
         )
+
+    @classmethod
+    def create_tsne_plot(cls):
+        tsne_plot_data = DiscoveryPersistence.get_session_tsne_plot_data()
+        if not tsne_plot_data:
+            raise PlotDataNotFoundException("Cannot find data to create plot!")
+
+        plot_df = tsne_plot_data.features_df.copy()
+
+        plot_df['is_train_data'] = 'Predicted'
+        plot_df.loc[tsne_plot_data.label_index, 'is_train_data'] = 'Labelled'
+
+        plot_df['Utility'] = -np.inf
+        plot_df.loc[tsne_plot_data.nolabel_index, 'Utility'] = pd.Series(tsne_plot_data.utility).values
+        plot_df = plot_df.sort_values(by='Utility', ascending=False)
+
```

Comment:
Duplicated code from experiment_postprocessor. Could it be useful to use ExperimentPostprocessor.plot_input_space() instead?

Reply:
hmmm... I will check it. I think the dulicated code is an artifact of the former implementation. I will delete the code for one of the two. I think, that the Postprocessor contained a fix which was on main but which I did not yet move to my logic. So I will fix that as well. Maybe you can also check but I think I was missing the normalizing logic in my implementation?

Reply:
Btw: I think I will keep the logic in discovery_service since it is actually not anymore part of the postprocessing of the experiment but a dedicated request which is first received by the service.